modules/engines/engine-hdi/src/main/java/com/sap/xsk/hdb/ds/processors/hdi/XSKGrantPrivilegesDefaultRoleProcessor.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2022 SAP SE or an SAP affiliate company and XSK contributors
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, v2.0
+ * which accompanies this distribution, and is available at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * SPDX-FileCopyrightText: 2022 SAP SE or an SAP affiliate company and XSK contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package com.sap.xsk.hdb.ds.processors.hdi;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.Arrays;
+
+public class XSKGrantPrivilegesDefaultRoleProcessor extends XSKHDIAbstractProcessor {
+
+  public void execute(Connection connection, String container, String user, String[] deployPaths) throws SQLException {
+    for (String path : Arrays.asList(deployPaths)) {
+      if (path.endsWith("xsk_technical_privileges.hdbrole"))  {
+        if (user == null) {
+          throw new IllegalStateException("xsk_technical_privileges.hdbrole assignment failed. No user provided.");
+        } else {
+          grantPrivileges(connection, container, user);
+        }
+        return;
+      }
+    }
```

Comment:
You can replace the for loop with streams:

```java
Arrays.stream(deployPaths).filter(deployPath -> deployPath.endsWith("hdbRoleFileName")).forEach(filePath -> grantPrivileges(connection, container, user));
```

and move user check outside the loop in the beginning of the method.